tests/camel-itest/src/test/java/org/apache/camel/itest/tx/JtaRouteTest.java

```diff
@@ -38,23 +65,94 @@ protected RouteBuilder createRouteBuilder() throws Exception {
             public void configure() throws Exception {
                 context.getRegistry().bind("PROPAGATION_REQUIRES_NEW", new RequiresNewJtaTransactionPolicy());
 
-                from("direct:requires_new")
+                from("direct:split_test")
                         .transacted("PROPAGATION_REQUIRES_NEW")
+                        .to("direct:split");
+
+                from("direct:split")
                         .split(body()).delimiter("_").to("direct:splitted").end()
-                        .log("after splitter log which you will never see...")
-                        .transform().constant("requires_new");
+                        .log("after splitter log which you will never see...");
 
                 from("direct:splitted").to("mock:splitted");
+
+                from("direct:multicast_test").routeId("r.route1")
+                        .log(LoggingLevel.DEBUG, "Entering route: ${routeId}")
+                        .transacted()
+                        .to("direct:multicast")
+                        .log("will never get here");
+
+                from("direct:multicast").routeId("r.route2")
+                        .log(LoggingLevel.DEBUG, "Entering route: ${routeId}")
+                        .multicast()
+                        .to("log:r.test", "direct:route3", "mock:test")
+                        .end();
+
+                from("direct:route3").routeId("r.route3")
+                        .log(LoggingLevel.DEBUG, "Entering route: ${routeId}");
```

Comment:
I think we should add a inlined processor that tests that the exchange is transacted, in all of these sub routes (not only here). This will help test the copied exchanges are marked as transactional.

Something ala
.process(e -> Assertions.assertTrue(e.isTransacted())

Reply:
OK, I get it and is there any way to do similar check with mock endpoint ?

Reply:
oh yeah you can do something with onExchangeReceived or what the name is

Reply:
@zhfeng can you work on this so we can get this PR ready and merged and backported as we want to build new releases of Camel LTS versions with this included.

Reply:
@davsclaus yeah, I'm working on it.

Reply:
Done